examples/python/xds/server.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Python implementation of the GRPC helloworld.Greeter server."""
+
+from concurrent import futures
+import argparse
+import logging
+import socket
+
+import grpc
+
+import helloworld_pb2
+import helloworld_pb2_grpc
+
+from grpc_reflection.v1alpha import reflection
+from grpc_health.v1 import health
+from grpc_health.v1 import health_pb2
+from grpc_health.v1 import health_pb2_grpc
+
+_DESCRIPTION = "A general purpose dummy server."
+
+
+class Greeter(helloworld_pb2_grpc.GreeterServicer):
+
+    def __init__(self, hostname):
+        self._hostname = hostname if hostname else socket.gethostname()
+
+    def SayHello(self, request, context):
+        return helloworld_pb2.HelloReply(
+            message=f"Hello {request.name} from {self._hostname}!")
+
+
+def serve(port, hostname):
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
+    helloworld_pb2_grpc.add_GreeterServicer_to_server(Greeter(hostname), server)
+    health_servicer = health.HealthServicer(
+        experimental_non_blocking=True,
+        experimental_thread_pool=futures.ThreadPoolExecutor(max_workers=4))
```

Comment:
Since examples tends to be copied a lot, maybe we should use a bigger number here? Will the threads become exhausted after 4 Watch RPC?

Reply:
optional: I felt that we should not publicize the `experimental_non_blocking` API at this point.

Reply:
Since we're using the non-blocking API, it shouldn't matter. I think even a single thread should suffice.

As for the appropriateness of the non-blocking API, I don't think we can advertise the health check API if we don't use it.